lib/segment/src/data_types/named_vectors.rs

```diff
@@ -67,7 +70,7 @@ impl<'a> NamedVectors<'a> {
     }
 
     pub fn keys(&self) -> impl Iterator<Item = &str> {
-        self.map.keys().map(|k| k.as_ref())
+        self.map.list.iter().map(|(k, _)| k.as_ref())
```

Comment:
why not use `.map.keys()`?

Reply:
Some strange lifetime error appeared. I got unresolved lifetimes locally

Reply:
with current approach it looks like abstraction leak